exporters/otlp/test/otlp_http_log_exporter_test.cc

```diff
@@ -251,7 +251,7 @@ TEST_F(OtlpHttpLogExporterTestPeer, ExportJsonIntegrationTest)
     report_span_id.assign(span_id_hex, sizeof(span_id_hex));
   }
 
-  ASSERT_TRUE(waitForRequests(8, old_count + 1));
+  ASSERT_TRUE(waitForRequests(30, old_count + 1));
```

Comment:
interesting. We started with 2, then 8, and now 30. Didn't know it's that slow with Valgrind :)

Reply:
I found otlp http client test under valgrind takes 6s most of the time. The `//sdk/test/_metrics:metric_instrument_test` take the longest time(almost 1minute). But sometime it takes more time to finish, maybe the CI node is busy at that time?

Reply:
Why do we need to increase the HTTP wait time if it's 6 secs most of the time as part of this PR? We should eventually remove the old metrics tests once new implementation is ready.

Reply:
This failure is happening every now and then in my PR [recently](https://github.com/open-telemetry/opentelemetry-cpp/runs/4827044509?check_suite_focus=true#step:5:134).

Reply:
We can merge this PR if that's the case. If the problem persists, we can think of mocking the HTTP responses instead of using a real server.

Reply:
Agree